hs_collection_resource/utils.py

```diff
@@ -30,3 +39,112 @@ def add_or_remove_relation_metadata(add=True, target_res_obj=None, relation_type
+    """
+    This func is to create a new csv file in bag that lists info of all contained resources
+    :param collection_obj: collection resource object
+    :param keep_local_csv: (default: False) if True, return the full path to csv on django server,
+    in which case user should be responsible for garbage collection (GC) of this csv file
+    :return: full path to csv file if keep_local_csv=True, otherwise None
+    """
+
+    tmp_dir = None
+    csv_full_path = None
+
+    try:
+        csv_full_name = CSV_FULL_NAME_TEMPLATE.format(collection_obj.short_id)
+
+        # remove all files in bag
+        for f in ResourceFile.objects.filter(object_id=collection_obj.id):
+            delete_resource_file_only(collection_obj, f)
+
+        if collection_obj.resources.count() > 0 or collection_obj.deleted_resources.count() > 0:
+            # prepare csv content
+            csv_content_list = []
+            # create headers
+            csv_header_row = ['Title',
+                              'Type',
+                              'ID',
+                              'URL',
+                              'Owners',
+                              'Sharing Status'
+                              ]
+            csv_content_list.append(csv_header_row)
+            # create rows for currently contained resources
+            for res in collection_obj.resources.all():
+                csv_data_row = [res.metadata.title,
+                                res.resource_type,
+                                res.short_id,
+                                RES_LANDING_PAGE_URL_TEMPLATE.format(res.short_id),
+                                _get_owners_string(list(res.raccess.owners.all())),
+                                _get_sharing_status_string(res)
+                                ]
+                csv_content_list.append(csv_data_row)
+
+            # create rows for deleted resources
+            for deleted_res_log in collection_obj.deleted_resources:
+                csv_data_row = [deleted_res_log.resource_title,
+                                deleted_res_log.resource_type,
+                                deleted_res_log.resource_id,
+                                DELETED_RES_STRING,
+                                _get_owners_string(list(deleted_res_log.resource_owners.all()))
+                                if deleted_res_log.resource_owners.count() > 0
+                                else DELETED_RES_STRING,
+                                DELETED_RES_STRING
+                                ]
+                csv_content_list.append(csv_data_row)
+
+            # create a new csv on django server
+            tmp_dir = tempfile.mkdtemp()
+            csv_full_path = tmp_dir + "/" + csv_full_name
+            with open(csv_full_path, 'w') as csv_file_handle:
+                w = csv.writer(csv_file_handle)
+                for row in csv_content_list:
+                    w.writerow(row)
```

Comment:
@pkdash yes, we always delete existing csv file and then re-create it from scratch, so the header row will be re-created as well each time.